### PR TITLE
mev support

### PR DESCRIPTION
**Pull request by nivBlox**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -181,3 +181,5 @@ replace github.com/prysmaticlabs/prysm => github.com/prysmaticlabs/prysm v1.4.2-
 replace github.com/google/flatbuffers => github.com/google/flatbuffers v1.11.0
 
 replace github.com/dgraph-io/ristretto => github.com/dgraph-io/ristretto v0.1.1-0.20211108053508-297c39e6640f
+
+replace github.com/bloxapp/ssv-spec v0.2.8-0.20230118104749-37a78c8e8a93 => github.com/nivBlox/ssv-spec v0.0.0-20230227123122-234200d9bd11
```
Comment (nivBlox) [inline go.mod:185]: should be removed once spec pr is merged

File: operator/duties/controller.go [hunk 2]
```diff
@@ -239,6 +243,41 @@ func (dc *dutyController) handleSlot(slot phase0.Slot) {
...
+}
+
+func (dc *dutyController) handleValidatorRegistration(slot phase0.Slot) {
+	// push if first time or every 10 epoch at first slot
+	epoch := dc.ethNetwork.EstimatedEpochAtSlot(slot)
+	firstSlot := dc.ethNetwork.GetEpochFirstSlot(epoch)
+	if dc.registered && (slot != firstSlot || uint64(epoch)%validatorRegistrationEpochInterval != 0) {
+		return
+	}
+	dc.registered = true
```
Comment (aaron-blox) [inline operator/duties/controller.go]: is this safe to set at this point? like if the next section errors, registered will still be true, is that okay?

File: protocol/v2/blockchain/beacon/beacon_mock.go [hunk 3]
```diff
@@ -28,12 +28,19 @@ func (b beaconMock) Events(ctx context.Context, topics []string, handler eth2cli
 	panic("implement me")
 }
 
-func (b beaconMock) GetBlindedBeaconBlock(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, graffiti, randao []byte) (*apiv1bellatrix.BlindedBeaconBlock, error) {
-	return nil, nil
+func (b beaconMock) SubmitValidatorRegistration(pubkey []byte, feeRecipient bellatrix.ExecutionAddress, sig phase0.BLSSignature) error {
```
Comment (aaron-blox) [inline protocol/v2/blockchain/beacon/beacon_mock.go:31]: if this is a mock? then why cant we just generate it from an interface(s)?

File: operator/duties/controller.go (hunk 2 above)
Comment (nivBlox) [reply operator/duties/controller.go]: removed

File: protocol/v2/blockchain/beacon/beacon_mock.go (hunk 3 above)
Comment (aaron-blox) [reply protocol/v2/blockchain/beacon/beacon_mock.go:31]: note: this can be changed to generated mock

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -40,7 +40,7 @@ Build stage Docker image:
     - docker tag $IMAGE_NAME:$CI_BUILD_REF $DOCKER_REPO_INFRA_STAGE:$CI_BUILD_REF
     - $DOCKER_LOGIN_TO_INFRA_STAGE_REPO && docker push $DOCKER_REPO_INFRA_STAGE:$CI_BUILD_REF
   only:
-    - stage
+    - mev-support
```
Comment (nivBlox) [inline .gitlab-ci.yml:43]: should revert

File: protocol/v2/ssv/validator/msgqueue_consumer.go [hunk 5]
```diff
@@ -133,7 +133,10 @@ func (v *Validator) GetLastHeight(identifier spectypes.MessageID) specqbft.Heigh
 	if r == nil {
 		return specqbft.Height(0)
 	}
-	return r.GetBaseRunner().QBFTController.Height
+	if ctrl := r.GetBaseRunner().QBFTController; ctrl != nil {
```
Comment (nivBlox) [inline protocol/v2/ssv/validator/msgqueue_consumer.go:136]: remove

File: protocol/v2/ssv/validator/startup.go [hunk 6]
```diff
@@ -24,11 +24,14 @@ func (v *Validator) Start() error {
 				continue
 			}
 			identifier := spectypes.NewMsgID(r.GetBaseRunner().Share.ValidatorPubKey, role)
-			if err := r.GetBaseRunner().QBFTController.LoadHighestInstance(identifier[:]); err != nil {
-				v.logger.Warn("failed to load highest instance",
-					zap.String("identifier", identifier.String()),
-					zap.Error(err))
+			if ctrl := r.GetBaseRunner().QBFTController; ctrl != nil {
```
Comment (nivBlox) [inline protocol/v2/ssv/validator/startup.go:27]: check if can remove

**Comment by aaron-blox** [comment]
continuing work in https://github.com/bloxapp/ssv/pull/870

**Comment by aaron-blox** [comment]
closing, continuing work in continuing work in https://github.com/bloxapp/ssv/pull/870

**Pull request closed without merging**